test/libsolidity/semanticTests/types/mapping/copy_from_mapping_to_mapping.sol

```diff
@@ -30,5 +30,5 @@ contract C {
 // ----
 // f() -> 0x20, 7, 8, 9, 0xa0, 13, 2, 0x40, 0xa0, 2, 3, 4, 2, 3, 4
 // gas irOptimized: 197102
-// gas legacy: 199887
-// gas legacyOptimized: 196845
+// gas legacy: 199827
+// gas legacyOptimized: 203694
```

Comment:
This one is odd. The only thing that changed in IR codegen is the replacement of `eq()` with `sub()` and both have the same cost (3 gas).

My guess would be that `sub()` prevents some optimizations from kicking in where they would for `eq()` (please verify this).

It this is the case then this change could be more costly than it appears unless we also tweak the optimizer. This might be another argument for switching to iteration against length instead of just changing the comparison.